base/main_test_bucket_pool.go

```diff
@@ -433,6 +445,44 @@ func (tbp *TestBucketPool) emptyPreparedStatements(ctx context.Context, b Bucket
 	}
 }
 
+// setXDCRBucketSetting sets the bucket setting "enableCrossClusterVersioning" for mobile XDCR
+func (tbp *TestBucketPool) setXDCRBucketSetting(ctx context.Context, bucket Bucket) {
+	if tbp.skipMobileXDCR {
+		tbp.Logf(ctx, "skipping mobile XDCR bucket setting")
+	}
+
+	store, ok := AsCouchbaseBucketStore(bucket)
+	if !ok {
+		tbp.Fatalf(ctx, "unable to get server management endpoints. Underlying bucket type was not GoCBBucket")
+	}
+
+	posts := url.Values{}
+	posts.Add("enableCrossClusterVersioning", "true")
+
+	url := fmt.Sprintf("/pools/default/buckets/%s", store.GetName())
+	_, statusCode, err := mgmtRequest(ctx, store, http.MethodPost, url, strings.NewReader(posts.Encode()))
+	if err != nil {
+		tbp.Fatalf(ctx, "request to mobile XDCR bucket setting failed, error: %v", err)
+	}
+	if statusCode != http.StatusOK {
+		tbp.Fatalf(ctx, "request to mobile XDCR bucket setting failed with status code, %d", statusCode)
+	}
+}
+
+// mgmtRequest sends REST API request to server
+func mgmtRequest(ctx context.Context, bucket CouchbaseBucketStore, method, url string, body io.Reader) ([]byte, int, error) {
```

Comment:
This is duplicated in xdcr package - I am tempted to make `MgmtRequest` API use this because it looks like all the callers are immediately reading output. This would mean that we don't have to worry about closing the body as the callers.

I can't think of a time right now where the management API would return a request that we'd want to handle streaming.